### PR TITLE
Bug 1538560 - [RFE]rename the project name mux-undefined

### PR DESCRIPTION
**Pull request by nhosoi**
mux-undefined ==> .orphaned
mux.sh: if there are any tests which create project "mux-undefined"
        just remove that project checking and creation.

### CODE REVIEW

File: docs/mux-logging-service.md [hunk 1]
```diff
@@ -164,8 +164,8 @@ example, something like this:
     done
 
 If the tag is not in the `project.namespacename` format, the log records will
-use a namespace of `mux-undefined`, so the log records will be in an index
-named something like `project.mux-undefined.$mux-undefined_uuid.YYYY.MM.DD`
+use a namespace of `.orphaned`, so the log records will be in an index
+named something like `project..orphaned.$mux-undefined_uuid.YYYY.MM.DD`
```
Comment (richm) [inline docs/mux-logging-service.md]: `.orphaned` isn't a namespace - @jcantrill - what is the full index name for orphaned records?  `.orphaned.orphan-uuid.YYYY.MM.DD`?

File: docs/mux-logging-service.md [hunk 2]
```diff
@@ -277,7 +277,7 @@ of the cluster.  It is expected that the record has the `@timestamp` field and
 is already formatted for the ViaQ data model.  If the tag matches
 `project.namespacename.**`, mux extracts the `namespacename` and passes this to
 the k8s meta filter plugin in order to get the namespace uuid.  If the tag does
-not match, mux uses a namespacename of `mux-undefined`.  Except for the k8s
+not match, mux uses a namespacename of `.orphaned`.  Except for the k8s
```
Comment (richm) [inline docs/mux-logging-service.md]: .orphaned isn't a namespace

File: docs/mux-logging-service.md [hunk 3]
```diff
@@ -360,7 +360,7 @@ The next stage in the pipeline is this filter:
 
 This examines the tag to see if it is of the form
 `project.namespacename.whatever`, and sets the field `mux_namespace_name` to
-the value of `namespacename` if it is in that form, or `mux-undefined`
+the value of `namespacename` if it is in that form, or `.orphaned`
```
Comment (richm) [inline docs/mux-logging-service.md]: .orphaned isn't a namespace

File: fluentd/configs.d/input-post-forward-mux.conf [hunk 4]
```diff
@@ -61,9 +61,9 @@
     @type record_transformer
     enable_ruby
     <record>
-      mux_namespace_name ${record['namespace_name'] || (tag_parts[0] == "project" && tag_parts[1]) || ENV["MUX_UNDEFINED_NAMESPACE"] || "mux-undefined"}
+      mux_namespace_name ${record['namespace_name'] || (tag_parts[0] == "project" && tag_parts[1]) || ENV["MUX_UNDEFINED_NAMESPACE"] || ".orphaned"}
       mux_need_k8s_meta ${(record['namespace_uuid'] || record.fetch('kubernetes', Hash.new)['namespace_id'].nil?) ? "true" : "false"}
-      kubernetes {"namespace_name":"${record['namespace_name'] || (tag_parts[0] == 'project' && tag_parts[1]) || ENV['MUX_UNDEFINED_NAMESPACE'] || 'mux-undefined'}","namespace_id":"${record['namespace_uuid'] || record.fetch('kubernetes', Hash.new)['namespace_id']}"}
+      kubernetes {"namespace_name":"${record['namespace_name'] || (tag_parts[0] == 'project' && tag_parts[1]) || ENV['MUX_UNDEFINED_NAMESPACE'] || '.orphaned'}","namespace_id":"${record['namespace_uuid'] || record.fetch('kubernetes', Hash.new)['namespace_id']}"}
```
Comment (richm) [inline fluentd/configs.d/input-post-forward-mux.conf]: @jcantrill what is the namespace_id for .orphaned?

File: test/mux.sh [hunk 5]
```diff
@@ -290,10 +290,10 @@ write_and_verify_logs() {
         os::cmd::try_until_success "oc get project testproj" 2>&1 | artifact_out
     elif [ $no_project_tag -eq 1 ]; then
         local myfield="SYSLOG_IDENTIFIER"
-        myproject=project.mux-undefined
+        myproject=project..orphaned
         espod=$es_pod
         # make sure this namespace exists
-        os::cmd::try_until_success "oc get project mux-undefined" 2>&1 | artifact_out
+        os::cmd::try_until_success "oc get project .orphaned" 2>&1 | artifact_out
```
Comment (richm) [inline test/mux.sh]: I thought this didn't work?

File: fluentd/configs.d/input-post-forward-mux.conf (hunk 4 above)
Comment (nhosoi) [reply fluentd/configs.d/input-post-forward-mux.conf]: fluent-plugin-kubernetes_metadata_filter/README.md:
* `orphaned_namespace_name` - The namespace to associate with records where the namespace can not be determined (default: `.orphaned`)
* `orphaned_namespace_id` - The namespace id to associate with records where the namespace can not be determined (default: `orphaned`)

Could there be a possibility record.fetch('kubernetes', Hash.new)['namespace_id'] is nil?   If so, we should add another default for orphane?
+      kubernetes {"namespace_name":"${record['namespace_name'] || (tag_parts[0] == 'project' && tag_parts[1]) || ENV['MUX_UNDEFINED_NAMESPACE'] || '.orphaned'}","namespace_id":"${record['namespace_uuid'] || record.fetch('kubernetes', Hash.new)['namespace_id'] || 'orphaned'}"}

File: test/mux.sh (hunk 5 above)
Comment (nhosoi) [reply test/mux.sh]: Right.  It fails...

File: fluentd/configs.d/input-post-forward-mux.conf (hunk 4 above)
Comment (richm) [reply fluentd/configs.d/input-post-forward-mux.conf]: > Could there be a possibility record.fetch('kubernetes', Hash.new)['namespace_id'] is nil? If so, we should add another default for orphane?

Yes.  If we cannot determine the namespace_id in any other way, we should use `orphaned`.

File: test/mux.sh (hunk 5 above)
Comment (richm) [reply test/mux.sh]: we don't need this check anymore

File: fluentd/configs.d/input-post-forward-mux.conf (hunk 4 above)
Comment (nhosoi) [reply fluentd/configs.d/input-post-forward-mux.conf]: https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/configs.d/input-post-forward-mux.conf#L87-L88
If mux_namespace_name is ENV["MUX_UNDEFINED_NAMESPACE"] or ".orphaned", do we want to set the tag to "mux" to skip "k8s meta"?

Comment (richm) [reply fluentd/configs.d/input-post-forward-mux.conf]: yes, good idea

**Comment by nhosoi** [comment]
Note: This pr is supposed to be released with https://github.com/openshift/openshift-ansible/pull/7792.

**Comment by richm** [comment]
does this PR work without the corresponding ansible change?  If so, what happens?

**Comment by nhosoi** [comment]
The corresponding openshift-ansible is for eliminating openshift_logging_mux_default_namespaces.  Functionality-wise, there is no impact with or without the openshift-ansible pr7792, I'd think.

**Comment by nhosoi** [comment]
> There is a backport 3.9 - do we need this in 3.9?

I set it just because the target release of bz1538560 is 3.9.z.  Probably, we could reset it to 3.10?

**Comment by richm** [comment]
> I set it just because the target release of bz1538560 is 3.9.z. Probably, we could reset it to 3.10?

Yes, I think so.  The other related change to rename logging to openshift-logging is also in 3.10 and is not backported.

**Comment by nhosoi** [comment]
Reset the target release of bz.  Removing backport/3.9, as well...

**Pull request merged**